source/features/reactions-avatars.tsx

```diff
@@ -49,31 +50,27 @@ function getParticipants(container: HTMLElement): Participant[] {
 }
 
 function add(): void {
-	for (const list of select.all('.has-reactions .comment-reactions-options:not(.rgh-reactions)')) {
-		const avatarLimit = arbitraryAvatarLimit - (list.children.length * approximateHeaderLength);
-
+	for (const list of select.all('.has-reactions:not(.rgh-reactions) .comment-reactions-options')) {
 		const participantByReaction = [...list.children as HTMLCollectionOf<HTMLElement>].map(getParticipants);
-		const flatParticipants = flatZip(participantByReaction, avatarLimit);
-
-		for (const {container, username, src} of flatParticipants) {
-			container.append(
-				<a>
-					<img src={src} />
-				</a>
-			);
-
-			// Without this, Firefox will follow the link instead of submitting the reaction button
-			if (!navigator.userAgent.includes('Firefox/')) {
-				(container.lastElementChild as HTMLAnchorElement).href = `/${username}`;
+		const participantsCount = participantByReaction.map(a => a.length).reduce((a, b) => a + b);
+		const rows = list.children.length >= 4 && participantsCount > 10 ? 2 : 1;
+		const columnCount = Math.ceil(list.children.length / rows);
+		const avatarReactionLimit = Math.round(avatarRowLimit / columnCount) - 1;
+
+		for (const participants of participantByReaction) {
+			for (const {container, username, src} of participants.slice(0, avatarReactionLimit)) {
+				container.append(
+					// Firefox doesn't handle links inside buttons correctly
+					<a href={isFirefox ? undefined : `/${username}`}>
+						<img src={src} />
+					</a>
+				);
 			}
 		}
 
-		list.classList.add('rgh-reactions');
+		list.closest('.has-reactions')!.classList.add('rgh-reactions');
 
-		// Overlap reaction avatars when near the avatarLimit
-		if (flatParticipants.length > avatarLimit * 0.9) {
-			list.classList.add('rgh-reactions-near-limit');
-		}
+		list.style.gridTemplateColumns = `repeat(${columnCount}, ${100 / columnCount}%)`;
```

Comment:
Perhaps this just needs to be

```suggestion
		list.style.gridTemplateColumns = `repeat(${columnCount}, auto)`;
```

And the columns will fit the content better (this is just an example, in this case it should still be 1 row):

<img width="703" alt="" src="https://user-images.githubusercontent.com/1402241/64595299-db2d7a00-d3db-11e9-9917-8abdc35e94ec.png">